src/lib/tls/tls_channel.h

```diff
@@ -34,10 +36,23 @@ class BOTAN_DLL Channel
    public:
       typedef std::function<void (const byte[], size_t)> output_fn;
       typedef std::function<void (const byte[], size_t)> data_cb;
-      typedef std::function<void (Alert, const byte[], size_t)> alert_cb;
+      typedef std::function<void (Alert)> alert_cb;
       typedef std::function<bool (const Session&)> handshake_cb;
       typedef std::function<void (const Handshake_Message&)> handshake_msg_cb;
+      static size_t IO_BUF_DEFAULT_SIZE;
 
+      Channel(const Callbacks& callbacks,
+              Session_Manager& session_manager,
+              RandomNumberGenerator& rng,
+              const Policy& policy,
+              bool is_datagram,
+              size_t io_buf_sz = IO_BUF_DEFAULT_SIZE);
+
+      /**
+       * DEPRECATED. This constructor is only provided for backward
+       * compatibility and should not be used in new implementations.
+       */
+      BOTAN_DEPRECATED("Use TLS::Channel(TLS::Callbacks ...)")
       Channel(output_fn out,
               data_cb app_data_cb,
               alert_cb alert_cb,
```

Comment:
There would not be much point in leaving this as a deprecated constructor if the signature changes due to change in alert_cb typedef. The compat constructor should continue to take a `std::function<void (Alert, const byte[], size_t)>` as the third param (regardless of `alert_cb`) and just bind nullptr/0 as the 2nd and 3rd arguments. Otherwise applications would still be forced to deal with an API change.